crates/ide_completion/src/completions/pattern.rs

```diff
@@ -135,10 +135,15 @@ fn pattern_path_completion(
                         .into_iter()
                         .for_each(|variant| acc.add_enum_variant(ctx, variant, None));
                 }
-                res @ (hir::PathResolution::TypeParam(_) | hir::PathResolution::SelfType(_)) => {
+                res @ (hir::PathResolution::TypeParam(_)
+                | hir::PathResolution::SelfType(_)
+                | hir::PathResolution::Def(hir::ModuleDef::Adt(hir::Adt::Struct(_)))) => {
```

Comment:
I think looking specifically for structs isn't going to work for all cases here, since unions and enums can also have completable associated types, and builtin types can also have them, for example here: https://doc.rust-lang.org/std/primitive.u32.html

So basically, the `PathResolution` doesn't matter for associated const completions, we just need to know its type and then call `iterate_path_candidates`. I'm not sure if there's a way to get the type without matching on all the variants of the `PathResolution` though. cc @Veykril

Reply:
There is no way to fetch type of a PathResolution as fas as I am aware